Services/CO.CDP.EntityVerification/Ppon/PponService.cs

```diff
@@ -1,9 +1,47 @@
+using System.Text;
+
 namespace CO.CDP.EntityVerification.Ppon;
 
 public class PponService : IPponService
 {
+    private readonly char[] CharPool = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".ToCharArray();
+    private readonly char[] EuansIdea = "ABCDEFGHIJKLMNPQRSTUVWXYZ".ToCharArray();
+    private readonly char[] EuansIdea2 = "123456789".ToCharArray();
+    private readonly Random RandomGenerator = new();
+
     public string GeneratePponId()
     {
-        return $"{Guid.NewGuid().ToString().Replace("-", string.Empty)}";
+        return $"{GenerateCodePattern()}-{GenerateMultiplier()}";
+    }
+
+    private string GenerateCodePattern()
+    {
+        var builder = new StringBuilder();
+        for (int i = 0; i < 12; i++)
+        {
+            char nextChar = (i / 4) % 2 == 0
+                        ? EuansIdea[RandomGenerator.Next(EuansIdea.Length)]
+                        : EuansIdea2[RandomGenerator.Next(EuansIdea2.Length)];
```

Comment:
Can we clean this up, and give these arrays meaningful names please?

Reply:
Done.